wgpu/src/lib.rs

```diff
@@ -2925,8 +3086,20 @@ pub struct BufferView<'a> {
 
 /// Write only view into mapped buffer.
 ///
+/// To get a `BufferViewMut`, first [map] the buffer, and then
+/// call `buffer.slice(range).get_mapped_range_mut()`.
+///
+/// `BufferViewMut` dereferences to `&mut [u8]`, so you can use all the usual
+/// Rust slice methods to access the buffer's contents. It also implements
+/// `AsMut<[u8]>`, if that's more convenient.
```

Comment:
the really important thing to note here is that it's not allowed to read from the buffer since for wgpu-core this may be arbitrarily slow (the memory might be write-combined) and the contents of the buffer are undefined

Reply:
The next paragraph says,

> It is possible to read the buffer using this view, but doing so is not recommended, as it is likely to be slow.

Is that not what you're looking for?

Reply:
ah, sorry, tunnelvision on changed lines

Right now we panic though on deref, right? (as per Matrix chat, we don't really want that, warning logged in debug mode at most)

Reply:
No, we only warn:

```rust
impl Deref for BufferViewMut<'_> {
    type Target = [u8];

    fn deref(&self) -> &Self::Target {
        if !self.readable {
            log::warn!("Reading from a BufferViewMut is slow and not recommended.");
        }

        self.data.slice()
    }
}
```